File: 2/2441/LargestPositiveIntegerThatExistsWithItsNegative.cpp
/*
https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/?envType=daily-question&envId=2024-05-02
*/

class Solution {
public:
    int findMaxK(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int sz=nums.size();
        for(int i=sz-1; i>=0; --i){
            int curr=nums[i];
            auto itr=find(nums.begin(), nums.end(), curr*-1);
            if(itr!=nums.end()) return curr;
        }
        return -1;
    }
};

File: 2/2441/LargestPositiveIntegerThatExistsWithItsNegative.py
# https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/?envType=daily-question&envId=2024-05-02

class Solution:
    def findMaxK(self, nums: List[int]) -> int:

        def customSearch(nums, val):
            left=0
            right=len(nums)-1
            while(left<=right):
                if nums[left]==-val:
                    return True
                elif nums[right]==-val:
                    return True
                left+=1
                right-=1
            return False

        nums.sort()
        sz=len(nums)
        for i in range(sz-1, -1, -1):
            curr=nums[i]
            if curr>=0:
                flag=customSearch(nums, curr)
                if flag==True:
                    return curr
        return -1